Bugfix/list of dicts support

austinwiltshire opened the pull request:
Passing in a list of dicts for the parameter space resulted in cv_results_ not being able to be converted to a dataframe. This was due to some parameter fit runs not always using all parameters, such as the case with sklearn's SVM's different kernels.

Using list.extend only actually added things to a list in the cv_results dictionary when the field was there, and the lists often got out of sync with each other, making the resulting dictionary of lists unable to be consumed by pandas.DataFrame

This uses the approach found in sklearn.gridsearch of masked arrays, and keeping track of the index we're actively filling in the resulting dictionary of masked arrays to keep everything aligned.

austinwiltshire committed:
Match gridsearch masked array approach

austinwiltshire committed:
Fix lines too long

CoitThomas commented on skopt/searchcv.py:
```diff
@@ -638,13 +652,34 @@ def fit(self, X, y=None, groups=None, callback=None):
+        self.cv_results_ = defaultdict(
+            partial(
+                MaskedArray,
+                np.empty(n_candidates,),
```
this syntax looks new and scary to me

CoitThomas reviewed (approved):
Looks good to me. Is there a unit test related to what you've refactored to verify the desired results?

austinwiltshire commented:
Unfortunately scikit optimize isn't being maintained and their unit tests don't appear to run.

I'll be exercising the code inside ensemble though in tests there, is that good enough @CoitThomas ?

austinwiltshire committed:
Fix typo from fixing lines

austinwiltshire committed:
Remove confusing tuple args

CoitThomas commented:
Yeah, if ensemble tests are able to verify then that's good enough for me 👍

holgern committed:
Fill user guide with some content

holgern committed:
Fix issue #877

* Use _transpose_list_array helper function in space to reduce code
* Use eps value also in inverse_transform in Normalize
* Clip inverse_transform also for Integer

holgern committed:
Revert changes in unit tests

holgern committed:
Fix doc warnings and add entry to whats_new

holgern committed:
Fix doc warning

holgern committed:
Fix whats_new

austinwiltshire committed:
Merge branch 'master' of https://github.com/scikit-optimize/scikit-optimize into bugfix/list_of_dicts_support

DavidWalz committed:
remove outdated version number & fix links to milestones page

holgern committed:
Merge branch 'master' into fix_issue_877

holgern committed:
Add link to development

holgern committed:
Merge pull request #880 from holgern/fix_issue_877

Try to fix issue with Integer out of bounds in transform

holgern committed:
Merge pull request #870 from holgern/Improve_user_guide

[WIP] Improve documentation

holgern committed:
Merge pull request #889 from DavidWalz/master

readme & docs: fix outdated version number and milestone links

tlienart committed:
Update bayesian-optimization.py

tlienart committed:
make pep8 happy

holgern committed:
Merge pull request #892 from tlienart/master

Fix maths display in bayesian_optimization doc

andreyvelich committed:
acquisition fix

holgern committed:
Merge pull request #896 from andreyvelich/fix-doc-spelling

Fix small spelling mistake

kartikayyer committed:
Fix position of minimum line in plot_objective()

Problem occurred when plot_dims is not the first
N dimensions

kartikayyer committed:
Add show_points and cmap options to plot_objective

`show_points` allows user to decide whether to see
evaluated points
`cmap` allows user to choose color map for contour
plot

kartikayyer committed:
Fix xlim of diagonals in plot_objective

kartikayyer committed:
Fix setting xlim with categorical spaces

holgern committed:
Merge pull request #898 from kartikayyer/master

[MRG] Add options to plot_objective

holgern committed:
Update minimum requirements

Change minimum requirements, so that they are compmatible with sklearn 0.23

holgern committed:
Update setup.py

holgern committed:
Update doc

holgern committed:
Merge pull request #906 from holgern/change_min_sklearn_version

[MRG] Update minimum requirements

austinwiltshire committed:
Merge branch 'master' into bugfix/list_of_dicts_support

austinwiltshire committed:
Merge Master

Two tests failed, I cleaned them up by making both tests more aware
of MaskedArray usage. Using MaskedArrays is more in line with how
sklearn's cross validated searches work too, so it better matches
their API.
